terminus-ui/paginator/src/paginator.component.spec.ts

```diff
@@ -1,470 +1,309 @@
-      component.ngAfterViewInit();
+      const titleEl = fixture.debugElement.query(By.css('.qa-paginator-current-page-menu .c-button__content')).nativeElement as HTMLElement;
+      expect(titleEl.textContent).toContain('1 - 20 of 100');
 
-      expect(component.isFirstPage(0)).toEqual(true);
     });
 
+    test(`should be hidden when set`, () => {
+      const fixture = createComponent(TestComponents.RecordsPerPage);
+      const hostComponent = fixture.componentInstance;
+      hostComponent.showRecordsPerPageSelector = false;
+      fixture.detectChanges();
 
-    test(`should return FALSE when the passed in page is not 0`, () => {
-      component.totalRecords = 125;
-      component.ngAfterViewInit();
-
-      expect(component.isFirstPage(7)).toEqual(false);
-      expect(component.isFirstPage(1)).toEqual(false);
-      expect(component.isFirstPage(-2)).toEqual(false);
+      expect(hostComponent.showRecordsPerPageSelector).toEqual(false);
+      expect(fixture.debugElement.query(By.css('.qa-paginator-per-page-select'))).toBeFalsy();
     });
-
   });
 
-
-  describe(`isLastPage()`, () => {
-
-    test(`should return TRUE when the page is the last in the array`, () => {
-      component.totalRecords = 20;
-      component.ngAfterViewInit();
-
-      expect(component.isLastPage(1)).toEqual(true);
-    });
-
-
-    test(`should return FALSE when the page is not the last in the array`, () => {
-      component.totalRecords = 20;
-      component.ngAfterViewInit();
-
-      expect(component.isLastPage(0)).toEqual(false);
-    });
-
-
-    test(`should return FALSE if the pagesArray doesn't exist`, () => {
-      component.pagesArray = undefined as any;
-
-      expect(component.isLastPage(1)).toEqual(false);
-    });
-
-
-    test(`should return TRUE when the page is the last in the array and not zero-based`, () => {
-      component.totalRecords = 20;
-      component.isZeroBased = false;
-      component.ngAfterViewInit();
-
-      expect(component.isLastPage(2)).toEqual(true);
-    });
+  describe(`clicking page buttons`, () => {
+
+    test(`should go to the next page`, fakeAsync(() => {
+      const fixture = createComponent(TestComponents.RecordsPerPage);
+      const hostComponent = fixture.componentInstance;
+      fixture.detectChanges();
+      const dir = 'next';
+      tick(2000);
+      clickForResults(fixture, dir);
+
+      const titleEl = fixture.debugElement.query(By.css('.qa-paginator-current-page-menu .c-button__content')).nativeElement as HTMLElement;
+      const npIndex = hostComponent.paginatorComponent.nextPageIndex;
+
+      expect(titleEl.textContent).toContain('11 - 20 of 100');
+      expect(npIndex).toEqual(1);
+    }));
+
+    test(`should go to the previous page`, fakeAsync(() => {
+      const fixture = createComponent(TestComponents.RecordsPerPage);
+      fixture.detectChanges();
+      const dir = 'previous';
+      tick(2000);
+      clickForResults(fixture, 'next');
+      clickForResults(fixture, dir);
```

Comment:
Can we assert something in between these two calls? We are kind of changing something, then changing it back, then verifying it never changed. So it seems we could remove both `clickForResults` calls and the test would still pass.

Or possibly just use a TestHost that has the pagination defaulted to something other than then first page.